ompi/mca/coll/sync/coll_sync_module.c

```diff
@@ -156,13 +203,26 @@ int mca_coll_sync_module_enable(mca_coll_base_module_t *module,
         OBJ_RETAIN(s->c_coll.coll_ ## name ## _module);  \
     }
 
-    CHECK_AND_RETAIN(bcast);
-    CHECK_AND_RETAIN(gather);
-    CHECK_AND_RETAIN(gatherv);
-    CHECK_AND_RETAIN(reduce);
-    CHECK_AND_RETAIN(reduce_scatter);
-    CHECK_AND_RETAIN(scatter);
-    CHECK_AND_RETAIN(scatterv);
+    if (0 != mca_coll_sync_component.barrier_before_nops ||
+        0 != mca_coll_sync_component.barrier_after_nops) {
+        CHECK_AND_RETAIN(alltoallv);
+        CHECK_AND_RETAIN(alltoallw);
+        CHECK_AND_RETAIN(bcast);
+        CHECK_AND_RETAIN(gather);
+        CHECK_AND_RETAIN(gatherv);
+        CHECK_AND_RETAIN(reduce);
+        CHECK_AND_RETAIN(reduce_scatter);
+        CHECK_AND_RETAIN(scatter);
+        CHECK_AND_RETAIN(scatterv);
+    } else {
```

Comment:
again, perhaps just a close bracket instead of the 'else', and some protection to only CHECK_AND_RETAIN alltoall[v|w] once if it goes through both sections of code.